rotors_gazebo_plugins/src/gazebo_multimotor_plugin.cpp

```diff
@@ -0,0 +1,217 @@
+#include "rotors_gazebo_plugins/gazebo_multimotor_plugin.h"
+
+namespace gazebo {
+
+void GazeboMultimotorPlugin::Load(physics::ModelPtr _model, sdf::ElementPtr _sdf) {
+  // Store the pointer to the model.
+  model_ = _model;
+  world_ = model_->GetWorld();
+  namespace_.clear();
+
+  //==============================================//
+  //========== READ IN PARAMS FROM SDF ===========//
+  //==============================================//
+
+  if (_sdf->HasElement("robotNamespace")) {
+    namespace_ = _sdf->GetElement("robotNamespace")->Get<std::string>();
+  } else {
+    gzerr << "[multimotor_plugin] Please specify a robotNamespace.\n";
+  }
+
+  node_handle_ = gazebo::transport::NodePtr(new transport::Node());
+
+  // Initialise with default namespace (typically /gazebo/default/)
+  node_handle_->Init();
+
+  getSdfParam<std::string>(_sdf, "actuatorCommandSubTopic", command_actuator_sub_topic_,
+                           command_actuator_sub_topic_);
+  getSdfParam<std::string>(_sdf, "actuatorStatePubTopic", motor_state_pub_topic_,
+                           motor_state_pub_topic_);
+
+  //=============================================//
+  //========== LOAD ROTORS AND SERVOS ===========//
+  //==============================================//
+  std::string joint_name, link_name;
+
+  // Add rotors.
+  if (_sdf->HasElement("rotors")) {
+    sdf::ElementPtr motors = _sdf->GetElement("rotors");
+    sdf::ElementPtr motor = motors->GetElement("rotor");
+
+    while (motor) {
+      // Only load valid motors
+      if (IsValidLink(motor) & IsValidJoint(motor)) {
+        motors_.push_back(std::make_unique<MotorModelRotor>(model_, motor));
+        gzdbg << "[gazebo_multimotor_plugin] Loaded rotor!\n";
+      } else {
+        gzdbg << "[gazebo_multimotor_plugin] Failed to load rotor!\n";
+      }
```

Comment:
Jenkins fails on 16.04 because std::make_unique is apparently a feature of c++14?

> error: 'make_unique' is not a member of 'std'

@michaelpantic what would be the right way to do this?